MantidQt/CustomInterfaces/src/DynamicPDF/DisplayCurveFitTest.cpp

```diff
@@ -0,0 +1,98 @@
+// includes for workspace handling
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/MatrixWorkspace.h"
+// includes for interface functionality
+#include "MantidQtCustomInterfaces/DynamicPDF/DisplayCurveFitTest.h"
+#include "MantidQtMantidWidgets/DisplayCurveFit.h"
+#include "MantidQtCustomInterfaces/DynamicPDF/DisplayCurveFitTest.h"
+#include "MantidQtMantidWidgets/RangeSelector.h"
+
+namespace {
+Mantid::Kernel::Logger g_log("DynamicPDF");
+}
+
+namespace MantidQt {
+namespace CustomInterfaces {
+namespace DynamicPDF {
+
+// Add this class to the list of specialised dialogs in this namespace only if
+// compiling in Debug mode
+//#ifndef NDEBUG
+DECLARE_SUBWINDOW(DisplayCurveFitTest)
+//#endif
+
+using curveType = MantidQt::MantidWidgets::DisplayCurveFit::curveType;
+using dcRange = MantidQt::MantidWidgets::DisplayCurveFit::dcRange;
+
+//           ++++++++++++++++++++++++++++
+//           +++++  Public Members  +++++
+//           ++++++++++++++++++++++++++++
+
+/// Constructor
+DisplayCurveFitTest::DisplayCurveFitTest(QWidget *parent)
+    : UserSubWindow{parent} {}
+
+DisplayCurveFitTest::~DisplayCurveFitTest() {}
+
+/**
+ * @brief Initialize the widgets defined within the form generated in
+ * Qt-Designer. Also, defined the SIGNALS to SLOTS connections.
+ */
+void DisplayCurveFitTest::initLayout() {
+  m_uiForm.setupUi(this);
+  connect(m_uiForm.dataSelector, SIGNAL(dataReady(const QString &)), this,
+          SLOT(loadSpectra(const QString &)));
+}
+
+//           +++++++++++++++++++++++++++
+//           +++++  Private Slots  +++++
+//           +++++++++++++++++++++++++++
+
+/**
+ * @brief The test proper that loads the fit curves to be
+ * displayed and the two ranges.
+ * @param workspaceName the name of the workspace containing
+ * the data of the curves to be displayed.
+ */
+void DisplayCurveFitTest::loadSpectra(const QString &workspaceName) {
+  auto workspace = Mantid::API::AnalysisDataService::Instance()
+                       .retrieveWS<Mantid::API::MatrixWorkspace>(
+                           workspaceName.toStdString());
+  if (workspace->getNumberHistograms() < 4) {
```

Comment:
Here `workspace` is not checked to be `!= nullptr` before being used.
I can imagine that coverity is going to complain about this "potential null dereference".